frame/election-provider-multi-phase/test-staking-e2e/src/mock.rs

```diff
@@ -594,18 +617,63 @@ pub fn roll_to(n: BlockNumber, delay_solution: bool) {
 	}
 }
 
+// Progress to given block, triggering session and era changes as we progress and ensuring that
+// there is a solution queued when expected.
+pub fn roll_to_with_ocw(n: BlockNumber, pool: Arc<RwLock<PoolState>>, delay_solution: bool) {
+	for b in (System::block_number()) + 1..=n {
+		System::set_block_number(b);
+		Session::on_initialize(b);
+		Timestamp::set_timestamp(System::block_number() * BLOCK_TIME + INIT_TIMESTAMP);
+
+		ElectionProviderMultiPhase::on_initialize(b);
+		ElectionProviderMultiPhase::offchain_worker(b);
+
+		if !delay_solution && pool.read().transactions.len() > 0 {
+			// build and submit extrinsics queued in the pool by ocw.
+			let encoded = pool.read().transactions[0].clone();
+			let extrinsic = Extrinsic::decode(&mut &*encoded).unwrap();
+			let call = match extrinsic.call {
+				RuntimeCall::ElectionProviderMultiPhase(call @ Call::submit_unsigned { .. }) =>
```

Comment:
maybe something like this would be simpler? 
`assert!(matches!(extrinsic.call), RuntimeCall::ElectionProviderMultiPhase(Call::submit_unsigned { .. }))`

Reply:
I changed the arms expressions to make sure we handle all the potential extrinsics queued in the pool instead of trying to decode only the first item in the transaction pool. I will add the assert! as well.